File: frontend/src/domain/Reader.ts
import type { Bookshelf } from "./Bookshelf";

export type Role = "READER" | "ADMIN";

export interface Reader {
  id: string;
  username: string;
  displayName: string;
  email: string;
  role: Role;
  defaultBookshelf: Bookshelf;
}

File: backend/src/main/kotlin/fr/amory/libris/library/domain/reader/Reader.kt
package fr.amory.libris.library.domain.reader

import fr.amory.libris.library.domain.bookshelf.BookshelfId

data class Reader(
    val id: ReaderId,
    val username: String,
    val email: String,
    val displayName: String,
    val defaultBookshelfId: BookshelfId,
)
